test/unit/services/JobRoleServiceTests.ts

```diff
@@ -43,22 +54,23 @@ describe('JobRoleService', function () {
 
     describe('getAllJobRoles', function () {
       it('should return all Job Roles from response', async () => {
-        const data = [jobRoleResponse];
+        const data = jobRoleResponseWrapper;
 
-        mock.onGet(`${config.API_URL}job-roles`).reply(200, data);
+        mock.onGet(`${config.API_URL}job-roles?page=1&limit=10`).reply(200, data);
 
-        const results = await getJobRoles();
-        
+        const results = await getJobRoles(1,10);
+        const { jobRoles, pagination } = results;
+
+        jobRoles[0].closingDate = new Date(jobRoles[0].closingDate);
 
-        results[0].closingDate = new Date(results[0].closingDate);
-        expect(results[0]).to.deep.equal(jobRoleResponse);
+        expect(results).to.deep.equal(jobRoleResponseWrapper);
       })
 
```

Comment:
You need to have tests that have a different page number and different limits for all scenarios - e.g the happy path and the two exceptions.

Reply:
Added